### ISSUE TITLE
ComboBox single-select: after selection is done, input field gets focus

### ISSUE THREAD

**Issue by topical** [tailwindlabs/headlessui#3027]
**What package within Headless UI are you using?**

@headlessui/vue

**What version of that package are you using?**

v1.7.19

**What browser are you using?**

Safari and Firefox

**Reproduction URL**

See #2795

**Describe your issue**

If you have a single-select ComboBox and select an option, the option list closes but the input element receives focus.

This has various implications:
* in terms of user experience, it looks like selection hasn't been finished (caret is blinking for non-obvious reasons)
* if mouse was used to open option list and select an option, it's unclear for user why input element gets focus out of a sudden
* with virtual keyboards, this behaviour is even more irritating, so #2272 tries to detect that, not setting focus in this case 
* detection of virtual keyboards is not safe, see remarks at #2272
* if ComboBox is the only input element on page, there is a blinking caret once user has finished selection, and there is no way to get rid of the caret; this is ugly
* for some applications, this behaviour even breaks application logic (see #2795)

Actually, I don't see any use case of this "set focus on input element" behaviour for single-select lists.

I suggest one of the following solutions:

* make this focus behaviour optional (actually, for mobiles it is already disabled)
* disable this behaviour completely for single-select lists

**Comment by RobinMalfait** [comment]
Hey!

We made some improvements to the Combobox component for Headless UI v2, this is currently not yet implemented in Vue yet, but you can play with the behavior in the React version. E.g.: open this on your phone: https://headlessui-react.vercel.app/combobox/combobox-countries

The main reason we re-focus the `ComboboxInput` is that you can continue from the current point, and go to the next field if there are any. If we don't re-focus the input then nothing would be focused (which typically means that the `body` is focused). If you use <kbd>tab</kbd> on desktop or use assistive technology on mobile then you would have to start from the top of the form again (because the `body` had focus).

Now, I do agree that it feels weird that the keyboard closes, it re-focuses the `combobox` and it opens your virtual keyboard again on mobile.

The improvements we made basically means that we never leave the focus of the `Combobox`. So there is no flash of your keyboard hiding and showing again. This behavior feels much better (similar to how you can type in a normal `input` and once you are done typing you are still focused in the `input` and you can continue to the next field if you want).

So to summarise, right now we never move focus away from the `ComboboxInput` which feels much better from a UX perspective _and_ solves a lot of issues that are no issues anymore.

Hope this helps!

**Comment by hudovisk** [comment]
Hey @RobinMalfait! We had the same issue after upgrading to headless ui v2 and I thought about a better pattern here would be to focus on the ComboboxButton after one option is selected for the single-select use case, that way the tab order is not lost and there is also opportunity to trigger the options list from keyboard. I made a hacky solution with setTimeout and references to button and input but would be great to have that behavior as an option in the Combobox component.
